src/overlays/actors/ovl_Obj_Y2shutter/z_obj_y2shutter.c

```diff
@@ -28,24 +43,158 @@ const ActorInit Obj_Y2shutter_InitVars = {
     (ActorFunc)ObjY2shutter_Draw,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80B9AEB0[] = {
+ShutterInfo sShutterInfo[] = {
+    // rectangular shutter door, activated by floor plate
+    {
+        object_kaizoku_obj_DL_0032A0,
+        &object_kaizoku_obj_Colheader_0035B0,
+        120.0f,
+        20.0f,
+        3.0f,
+        -20.0f,
+        3.0f,
+        4,
+        8,
+        0xA0,
+    },
+    // hexagonal shutter, activated by hitting crystal switch
+    {
+        object_kaizoku_obj_DL_005720,
+        &object_kaizoku_obj_Colheader_005EC8,
+        150.0f,
+        1.0f,
+        0.04f,
+        -1.0f,
+        0.04f,
+        6,
+        0xC,
+        0xA0,
+    },
+};
+
+static InitChainEntry sInitChain[] = {
     ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneScale, 800, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneDownward, 800, ICHAIN_STOP),
 };
 
-#endif
+void ObjY2shutter_Init(Actor* thisx, PlayState* play) {
+    s32 pad[2];
+    ShutterInfo* info;
+    ObjY2shutter* this = THIS;
+
+    info = &sShutterInfo[OBJY2SHUTTER_GET_SHUTTER_TYPE(this)];
```

Comment:
It would, but then I would be defining `this` after it's been used. As per a review on one of my other PRs, I'm making `OBJY2SHUTTER_GET_SHUTTER_TYPE` take the actor pointer directly instead of needing `this`, so I could pass it thisx, but I'm not sure if that would be bad form here

Reply:
If the stack has to go in this order there's nothing you can really do about that.